src/gvpn_controller.py

```diff
@@ -220,7 +227,7 @@ def serve(self):
                         data[48:54].encode("hex"), data[54:56].encode("hex")))
  
                     if not CONFIG["on-demand_connection"]:
-                        continue
+                        continue 
```

Comment:
I think you added whitespace at the end of the line. Be careful about this, and consider adjusting your editor config to strip trailing newlines or to color trailing whitespace.